### PR TITLE
Add LDAP login support

### PR DESCRIPTION
**Pull request by ensky**
Provide a simple LDAP login method.
Just like github login, if LDAP user not exist, we'll create a new user for him.

### CODE REVIEW

**Comment by jespino** [comment]
I like it!, we have to test it, but looks great :smile:

File: taiga/base/connectors/ldap.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+from django.conf import settings
+from . import exceptions as exc
+
+SERVER = getattr(settings, "LDAP_SERVER", "")
+DN_FORMAT = getattr(settings, "LDAP_DN_FORMAT", "")
+BASE_EMAIL = getattr(settings, "LDAP_BASE_EMAIL", "")
+
+def login (username: str, password: str) -> tuple:
+	dn = DN_FORMAT.format(username=username)
```
Comment (bameda) [inline taiga/base/connectors/ldap.py]: Indent error

Comment (ensky) [inline taiga/base/connectors/ldap.py]: My bad, fixed :)

**Comment by jespino** [comment]
We are thinkg about moving it to a plugin system for authentication. we have to make some changes in the core code, but then it can be in something like taiga-contrib-auth-ldap python package and install it easy using pip and adding it to INSTALLED_APPS. Do you like the idea?

**Comment by jespino** [comment]
Hi @ensky, I made some changes in the refactor/auth-modules branch to enable login through external plugins, and I created the repository jespino/taiga-contrib-ldap-auth, for ldap authentication. We don't use LDAP and then, we can't maintain this plugin, if you want, I can transfer the repository to you (and of course, in the next release, link the plugin repository in our Readme).

**Comment by ensky** [comment]
Hi @jespino, thanks for this refactor, I think it is a good idea that taiga can support third-party authentication now. Yes you can transfer this project into my account and link it in your ReadMe. (Although it's also hard for me to maintain this plugin, but I think it'll be better not to put it in official account since you don't use it. Let the community do it.)

**Comment by stevenroose** [comment]
Perhaps the existence of the plugin could be mentioned somewhere so it is clear that integration with LDAP is possible :)

**Comment by superalex** [comment]
Your are right @stevenroose, I added a new section here https://taiga.io/support/ for community maintained plugins! :+1:

**Pull request closed without merging**